### ISSUE TITLE
Release 0.6.1: invalid conversion from ‘int’ to ‘packbuf_t’ error with g++

### ISSUE THREAD

**Issue by ajaypanyala** [flame/blis#374]
Am using Release 0.6.1 and trying to compile `https://github.com/flame/blis/blob/master/test/test_gemm.c` with `g++ versions > 7.x` (Have not tested with `g++ <= 6.x`)

```g++  -c test_gemm.cc -I.```

```
In file included from test_gemm.cc:36:0:

blis.h: In function ‘void bli_mem_clear(mem_t*)’:
blis.h:25086:32: error: invalid conversion from ‘int’ to ‘packbuf_t’ [-fpermissive]
  bli_mem_set_buf_type( -1, mem );
                                ^
blis.h:25052:13: note:   initializing argument 1 of ‘void bli_mem_set_buf_type(packbuf_t, mem_t*)’
 static void bli_mem_set_buf_type( packbuf_t buf_type, mem_t* mem )
             ^~~~~~~~~~~~~~~~~~~~
```

**Comment by devinamatthews** [comment]
@fgvanzee the Travis CI setup should probably include a C++-compilation test to catch these kind of bugs. The fix in this case is a simple cast (or using a valid enum constant).

**Comment by fgvanzee** [comment]
@ajaypanyala It would be very helpful to us if you could you use triple backquotes before and after your output so that it's formatted as fixed-width font. (Also, it appears your command line invocation was incomplete; I was fixated on the fact that I couldn't see how `g++` was even finding `blis.h`.)

**Comment by ajaypanyala** [comment]
@fgvanzee I din't realize the compile line got mixed up with the error. Edited my original post. Thanks!

**Comment by fgvanzee** [comment]
@ajaypanyala Thanks for that edit, Ajay. I'm guessing that you have the path to `blis.h` taken care of in some subtle way?

I trust Devin when he says this is an easy typecasting fix. If we can reproduce your issue, we can take care of the error your received as well as any others that would show up.

**Comment by ajaypanyala** [comment]
Yes, `blis.h` is in the same folder as the test code in my case.

**Comment by fgvanzee** [comment]
@ajaypanyala I think I've fixed your issue in d7a7679. Please give it a try.

Please be aware that you need other compiler flags in order to successfully compile the drivers in `test`. Those drivers were actually intended to be compiled via the `Makefile` in that directory. What I recommend is starting with a fresh/unmodified clone. Then:
```
$ ./configure auto; make      # Build how you normally would.
$ cd test
$ make test_gemm_blis.x
gcc -O3 -Wall -Wno-unused-function -Wfatal-errors -fPIC -std=c99
-D_POSIX_C_SOURCE=200112L -I../include/haswell -DBLIS_VERSION_STRING=\"0.6.1-2\"
-I. -DBLIS -c test_gemm.c -o test_gemm_blis.o
gcc test_gemm_blis.o ../lib/haswell/libblis.a -lm -lpthread -lrt -o test_gemm_blis.x
# Note the command line options used above. Copy and paste everything except
# change gcc to g++ and run that manually.
$ g++ -O3 -Wall -Wno-unused-function -Wfatal-errors -fPIC -std=c99
-D_POSIX_C_SOURCE=200112L -I../include/haswell -DBLIS_VERSION_STRING=\"0.6.1-2\"
-I. -DBLIS -c test_gemm.c -o test_gemm_blis.o
cc1plus: warning: command line option ‘-std=c99’ is valid for C/ObjC but not for C++
# You'll also need to manually link the .o and libraries to the .x. It shouldn't
# matter if you use gcc or g++ here.
$ g++ test_gemm_blis.o ../lib/haswell/libblis.a -lm -lpthread -lrt -o test_gemm_blis.x
# And that should work without error now.
```
That will allow you to build `test_gemm.c` using `g++`.

**Comment by ajaypanyala** [comment]
@fgvanzee I can confirm that `d7a7679` works. Thanks!
